code/modules/admin/respawn_as_self.dm

```diff
@@ -0,0 +1,35 @@
+/client/proc/respawn_as_self()
+	set name = "Respawn as Self"
+	set desc = "Respawn instantly with currenly loaded character on place."
+	set category = "Special Verbs"
+
+	if(!check_rights(R_SPAWN))
+		return
+
+	var/input = ckey(input(src, "Specify which key will be respawned as human.", "Respawn as Human", "[usr.ckey]") as text)
```

Comment:
You could just select a ghost from the dead mob list instead of this faffing around with manually entering ckeys.